### PR TITLE
Allow installing casks with `brew install`.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [x] Have you successfully run `brew man` locally and committed any changes?

### CODE REVIEW

**Comment by miccal** [comment]
Fantastic work here @reitermarkus, thank you.

File: Library/Homebrew/cli/named_args.rb [hunk 1]
```diff
@@ -73,6 +86,18 @@ def to_resolved_formulae_to_casks
         end
       end
 
+      # Convert named arguments to `Tap`, `Formula` or `Cask` objects.
+      # If both a formula and cask exist with the same name, returns the
+      # formula and prints a warning unless `only` is specified.
+      def to_objects(only: nil)
+        @to_objects ||= {}
+        @to_objects[only] ||= downcased_unique_named.flat_map do |name|
+          next Tap.fetch(name) if only == :tap || (only.nil? && name.count("/") == 1 && !name.start_with?("./", "/"))
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cli/named_args.rb:95]: Could this get split into a few conditionals or variables? It's a bit hard to follow as a post-if.

File: Library/Homebrew/cmd/install.rb [hunk 2]
```diff
@@ -27,65 +119,25 @@ def install_args
...
-                          "run `./configure --help` and otherwise determine how to turn the software "\
-                          "package into a Homebrew package."
-      switch "-g", "--git",
-             description: "Create a Git repository, useful for creating patches to the software."
+
       conflicts "--ignore-dependencies", "--only-dependencies"
-      conflicts "--devel", "--HEAD"
       conflicts "--build-from-source", "--build-bottle", "--force-bottle"
+
+      formula_only_options.each do |options|
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/install.rb:131]: Could this be inlined in this function as `formula_only_options` isn't used elsewhere and it'd allow you to read through the list of options in order (as before)?

File: docs/Manpage.md [hunk 3]
```diff
@@ -337,16 +343,52 @@ installed formulae or, every 30 days, for all formulae.
...
+  Target location for Input Methods. Default: `~/Library/Input Methods`
+* `--internet_plugindir`:
+  Target location for Internet Plugins. Default: `~/Library/Internet Plug-Ins`
+* `--audio_unit_plugindir`:
+  Target location for Audio Unit Plugins. Default: `~/Library/Audio/Plug-Ins/Components`
+* `--vst_plugindir`:
+  Target location for VST Plugins. Default: `~/Library/Audio/Plug-Ins/VST`
+* `--vst3_plugindir`:
+  Target location for VST3 Plugins. Default: `~/Library/Audio/Plug-Ins/VST3`
+* `--screen_saverdir`:
```
Comment (MikeMcQuaid) [inline docs/Manpage.md:380]: This really bloats the `brew install` options for options that feel like they are both infrequently used and better handled with environment variables.

File: Library/Homebrew/cmd/install.rb (hunk 2 above)
Comment (reitermarkus) [reply Library/Homebrew/cmd/install.rb:131]: I had this before but RuboCop complained that the block is too long and I didn't want to fight it.

Comment (MikeMcQuaid) [reply Library/Homebrew/cmd/install.rb:131]: @reitermarkus Bah. I feel like ideally these usage blocks would be excluded automatically and, if not, we could exclude `cmd` and `dev-cmd` entirely
